pkg/engine/armresources_test.go

```diff
@@ -591,7 +591,7 @@ func TestGenerateARMResourcesWithVMSSAgentPool(t *testing.T) {
 	}
 }
 
-func getNICIPConfigs(n int) *[]network.InterfaceIPConfiguration {
+func getNICIPConfigs(n int, skipMasterLB bool) *[]network.InterfaceIPConfiguration {
```

Comment:
updated this convenience func so we could more easily UT scenarios that don't build a master LB (e.g., single master private cluster)

Reply:
Well, this was hear before I invited myself :)

But mainly:

https://raw.githubusercontent.com/Azure/azure-sdk-for-go/v36.2.0/services/network/mgmt/2018-08-01/network/models.go

Sorry, yeah, that file is literally too large for github to handle. Search for `type InterfacePropertiesFormat struct {` and look for the `IPConfigurations` property

Reply:
@jhendrixMSFT do you know why slices are `*[]type` rather than just `[]type`?

Reply:
@jackfrancis for expediency, I guess we should just leave it as is. Just seems like an unneeded indirection.

Reply:
Sorry for the delay.  It's so that we can send empty arrays over-the-wire (there are tests for this in the atuorest test suite).  The other option would be to remove the `,omitempty` from the tag however I'm not sure if this has other unwanted side-effects (I'm looking into this in our track 2 code generator).

Reply:
Thank you for the explanation, @jhendrixMSFT.

Seems like removing `,omitempty`, as you point out, would be equivalent functionality with less verbosity. +1 from me

Reply:
I looked into this a bit more, unfortunately it breaks the scenario where you need to differentiate between omitting the slice vs sending an empty slice (same goes for maps).  I think we're going to have to live with this.

I keep coming back to having a builder pattern that removes the need to create these struct literals.  I haven't had a chance to explore this in depth, and it might introduce other complexities that offset the gains for pointer-to-type.

Reply:
Thank you for jogging my memory. I forgot we had to support empty slices b/c they are significant to some services 😞 . 

Perhaps, those services could mark these aberrant fields in the OpenAPI spec with something like `"x-non-conforming-evil-ptr-slice-bc-my-service-needs-empty-slices": true`, and then all the rest of the APIs can be prettier?? What do you think??

Reply:
I like the idea however I think getting all services to correctly update their swaggers would be a hard sell.  Plus I think there are legitimate cases where you actually want to send the empty slice, so getting it right might be tricky.

Another option would be to maintain our own marshalling stack that applies the semantics we want.  We'd have to do it for both JSON and XML, and TBH I don't think the ends justify the cost (maybe it's cheaper than I envision).